File: mvp/test-fixtures/all-languages/src/models/Product.ts
export interface IProduct {
    id: number;
    name: string;
    price: number;
}

export class Product implements IProduct {
    id: number;
    name: string;
    price: number;
    stock: number;

    constructor(id: number, name: string, price: number, stock: number = 0) {
        this.id = id;
        this.name = name;
        this.price = price;
        this.stock = stock;
    }

    isAvailable(): boolean {
        return this.stock > 0;
    }

    updateStock(quantity: number): void {
        this.stock += quantity;
    }
}

File: mvp/test-fixtures/all-languages/src/models/Product.py
class Product:
    """Product model"""
    
    id: int
    name: str
    price: float
    stock: int
    
    def __init__(self, id: int, name: str, price: float, stock: int = 0):
        self.id = id
        self.name = name
        self.price = price
        self.stock = stock
    
    def is_available(self) -> bool:
        """Check if product is in stock"""
        return self.stock > 0
    
    def update_stock(self, quantity: int) -> None:
        """Update product stock"""
        self.stock += quantity
